### PR TITLE
Cart functionality

### LINKED ISSUE #50: Implement cart
To make it easier to use the SDK on clients, we should create the notion of cart, which encapsulates the order logic.
E.g: 
`Cart().addProduct({productId, quantity})` - Don't have to be authenticated
`Cart().removeProduct({productId, quantity})` - Don't have to be authenticated
`Cart().empty()` - Don't have to be authenticated
`Cart().removeProduct({productId, quantity})` - Don't have to be authenticated
`Cart().checkout(paymentMethodId, deliveryAddress, resumeOrderId)` - Have to be authenticated

### PR DESCRIPTION
**Pull request by lazreg87**
Implemented cart functionality.
We use the existing order flow and allow the user to call checkout, which wraps the order creation and pay. (requires that the user is authenticated)

Fixes #50

### CODE REVIEW

File: src/functionality/cart.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  get() {
+    return this._cart;
+  }
+
+  empty() {
+    this._cart = [];
+  }
+
+  addProduct({ productId, quantity }) {
```
Comment (tigo0) [inline src/functionality/cart.js]: Maybe add the whole product, then you can retrieve and show product details from the cart?

**Comment by tigo0** [comment]
How does this work with `sub_products`?
Assuming the `demo-store`, where you have `sneakers` as `main` products and `sizes` as `sub_products`. Not sure if you can order Puma size 6, where Puma is main product and size 6 is `sub_product` in this case?

File: src/functionality/cart.js (hunk 1 above)
Comment (lazreg87) [reply src/functionality/cart.js]: good idea

File: src/functionality/cart.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const exisitingProduct = this._findProductIndex(productId);
+    // If product item exists, we replace with new product item
+    if (exisitingProduct > -1) {
+      this._cart[exisitingProduct] = { productId, quantity };
```
Comment (NicolasBonduel) [inline src/functionality/cart.js]: Shouldn't this add up the quantities?

File: src/functionality/cart.js (hunk 1 above)
Comment (lazreg87) [reply src/functionality/cart.js]: Hm, you do risk showing stale products though. If you just store the ids locally, you could retrieve the whole cart from the API and be sure that the products are up-to-date.

File: src/functionality/cart.js (hunk 2 above)
Comment (lazreg87) [reply src/functionality/cart.js]: Didn't want to force any add/subtract logic, so you just replace it with the new quantity

**Pull request merged**